web-ng/src/app/components/observation-form/observation-form.component.ts

```diff
@@ -106,16 +110,27 @@ export class ObservationFormComponent {
   }
 
   onSave() {
-    const updatedResponses: Map<string, Response> = this.extractResponses();
-    const updatedObservation = this.observation!.withResponses(
-      updatedResponses
-    );
-    this.dataStoreService
-      .updateObservation(this.projectId!, updatedObservation)
-      .then(() => this.navigateToFeature(updatedObservation))
-      .catch(() => {
-        alert('Observation update failed.');
-      });
+    this.authService.getUser$().pipe(first()).subscribe(user => {
+      if (!user) {
+        throw Error('Login required to update observation.');
+      }
+      const lastModified = new AuditInfo(
+        user,
+        new Date(),
```

Comment:
Please add /* argname= */ comment or assign to a local var to clarify what this arg is.